ocs_ci/deployment/aws.py

```diff
@@ -134,18 +150,190 @@ def destroy_cluster(self, log_level="DEBUG"):
+                self.upi_script_path,
+            )
+            shutil.copy2(
+                os.path.join(bindir, 'oc'), self.upi_script_path
+            )
+
+        def deploy(self, log_cli_level='DEBUG'):
+            """
+            Exact deployment will happen here
+
+            Args:
+                log_cli_level (str): openshift installer's log level
+                    (default: "DEBUG")
+            """
+            logger.info("Deploying OCP cluster")
+            logger.info(
+                f"Openshift-installer will be using loglevel:{log_cli_level}"
+            )
+
+            # Invoke upi_on_aws-install.sh
+            cidir = os.getcwd()
+            logger.info("Changing CWD")
+            try:
+                os.chdir(self.upi_script_path)
+            except OSError:
+                logger.exception(
+                    f"Failed to change CWD to {self.upi_script_path} "
                 )
-        except Exception:
-            logger.error(traceback.format_exc())
+            logger.info(f"CWD changed to {self.upi_script_path}")
+
+            with open("./upi_on_aws-install.sh", "r") as fd:
+                buf = fd.read()
+            data = buf.replace("openshift-qe-upi", "ocs-qe-upi")
+            with open("./upi_on_aws-install.sh", "w") as fd:
+                fd.write(data)
+
+            sys.path.append(self.upi_script_path)
+            full_path = os.getcwd()
+            proc = Popen(
+                [f'{full_path}/'
+                 f'upi_on_aws-install.sh'],
+                stdout=PIPE, stderr=PIPE,
+            )
+            stdout, stderr = proc.communicate()
+
+            # Change dir back to ocs-ci dir
+            os.chdir(cidir)
+
+            if proc.returncode:
+                logger.error(stderr)
+                raise exceptions.CommandFailed("upi install script failed")
+            logger.info(stdout)
+
+    def deploy_ocp(self, log_cli_level='DEBUG'):
+        """
+        OCP deployment specific to AWS UPI
+
+        Args:
+             log_cli_level (str): openshift installer's log level
+                (default: 'DEBUG')
+        """
+        super(AWSUPI, self).deploy_ocp(log_cli_level)
+        volume_size = config.ENV_DATA.get('DEFAULT_EBS_VOLUME_SIZE', 100)
+        # existing function looks for terraform files
+        self.add_volume(volume_size)
+
+    def destroy_cluster(self, log_level="DEBUG"):
+        """
+        Destroy OCP cluster for AWS UPI
+
+        Args:
+            log_level (str): log level for openshift-installer (
+                default:DEBUG)
+        """
+        cluster_name = get_cluster_name(self.cluster_path)
+        self.destroy_volumes()
+        suffixes = ['vpc', 'inf', 'sg', 'bs', 'ma']
+        for i in range(3):  # TODO: read in num_workers
+            suffixes.append(f'no{i}')
+        stack_names = [f'{cluster_name}-{suffix}' for suffix in suffixes]
+        cf = boto3.client('cloudformation')
```

Comment:
Can we move this to ocs_ci/utility/aws.py and create function there like delete_claudformation_stack ?

Reply:
We can, however this was intended to be temporary as we will be calling the script in the repo that removes the stacks.

Reply:
Since this is no longer temporary, I'd like to address this again. I don't see the value in creating a wrapper function around what is essentially two lines of code. Creating the client, then executing boto's `delete_stack` method.

I'd like to avoid creating unnecessary wrapper utilities since we tend to run into many issues around supporting every use case the underlying methods support natively. `run_cmd` being a prime example of this.